File: frontend/src/api/audit.ts
import { apiClient } from './client'

export interface AuditEntry {
  id: string
  user_id?: string
  action: string
  entity_name: string
  entity_id?: string
  created_at: string
  payload?: Record<string, unknown>
}

export interface AuditListResponse {
  data: AuditEntry[]
  total: number
}

export async function listAuditLog(skip = 0, limit = 100): Promise<AuditListResponse> {
  const { data } = await apiClient.get<AuditListResponse>('/admin/audit-logs', {
    params: { skip, limit },
  })
  return data
}

File: backend/app/schemas/audit.py
from pydantic import BaseModel
from typing import Optional, Dict, Any, List
from datetime import datetime
import uuid

class AuditLogOut(BaseModel):
    id: uuid.UUID
    user_id: Optional[uuid.UUID]
    action: str
    entity_name: str
    entity_id: Optional[uuid.UUID]
    payload: Optional[Dict[str, Any]]
    created_at: datetime
    
    class Config:
        from_attributes = True

class AuditLogListOut(BaseModel):
    data: List[AuditLogOut]
    total: int
